File: room-demo-rxandroid-livedata/src/main/java/com/example/pavneet_singh/room_demo_rxandroid/util/DateRoomConverter.java
package com.example.pavneet_singh.room_demo_rxandroid.util;

import androidx.room.TypeConverter;

import java.util.Date;

/**
 * Created by Pavneet_Singh on 12/31/17.
 */

public class DateRoomConverter {

    @TypeConverter
    public static Date toDate(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long toLong(Date value) {
        return value == null ? null : value.getTime();
    }
}

File: room-demo-kotlin-mvvm-dagger/src/main/java/com/example/pavneet_singh/room_demo_kotin_mvvm_dagger/util/DateRoomConverter.kt
package com.example.pavneet_singh.room_demo_kotin_mvvm_dagger.util

import androidx.room.TypeConverter
import java.util.*

/**
 * Created by Pavneet_Singh on 2020-01-25.
 */

object DateRoomConverter {
    @TypeConverter
    @JvmStatic
    fun toDate(value: Long?): Date? {
        return value?.let { Date(it) }
    }

    @TypeConverter
    @JvmStatic
    fun toLong(value: Date?): Long? {
        return value?.time
    }
}